eclair-core/src/test/scala/fr/acinq/eclair/io/OpenChannelInterceptorSpec.scala

```diff
@@ -0,0 +1,153 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor
+import akka.actor.testkit.typed.scaladsl.{ScalaTestWithActorTestKit, TestProbe}
+import akka.actor.typed.ActorRef
+import akka.actor.typed.eventstream.EventStream
+import com.typesafe.config.ConfigFactory
+import fr.acinq.bitcoin.scalacompat.Crypto.PrivateKey
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Crypto, DeterministicWallet, Satoshi, SatoshiLong}
+import fr.acinq.eclair.blockchain.DummyOnChainWallet
+import fr.acinq.eclair.blockchain.fee.FeeratePerKw
+import fr.acinq.eclair.channel.{ChannelAborted, ChannelFlags, LocalParams}
+import fr.acinq.eclair.io.OpenChannelInterceptor.{DefaultParams, OpenChannelNonInitiator}
+import fr.acinq.eclair.io.Peer.{ChannelId, OutgoingMessage, SpawnChannelNonInitiator}
+import fr.acinq.eclair.io.PendingChannelsRateLimiter.AddOrRejectChannel
+import fr.acinq.eclair.wire.protocol.{Error, Init, NodeAddress, OpenChannel}
+import fr.acinq.eclair.{AcceptOpenChannel, CltvExpiryDelta, Features, InterceptOpenChannelCommand, InterceptOpenChannelPlugin, InterceptOpenChannelReceived, MilliSatoshiLong, RejectOpenChannel, TestConstants, UInt64, randomBytes32}
+import org.scalatest.Outcome
+import org.scalatest.funsuite.FixtureAnyFunSuiteLike
+import scodec.bits.ByteVector
+
+import scala.concurrent.duration.DurationInt
+
+class OpenChannelInterceptorSpec extends ScalaTestWithActorTestKit(ConfigFactory.load("application")) with FixtureAnyFunSuiteLike {
+  val fundingAmount: Satoshi = 1 sat
+  val publicKey: Crypto.PublicKey = PrivateKey(ByteVector32.One).publicKey
+  val remoteNodeId: Crypto.PublicKey = PrivateKey(ByteVector32.One).publicKey
+  val temporaryChannelId: ByteVector32 = ByteVector32.Zeroes
+  val openChannel: OpenChannel = OpenChannel(ByteVector32.Zeroes, temporaryChannelId, fundingAmount, 0 msat, 1 sat, UInt64(1), 1 sat, 1 msat, FeeratePerKw(1 sat), CltvExpiryDelta(1), 1, publicKey, publicKey, publicKey, publicKey, publicKey, publicKey, ChannelFlags.Private)
+  val localParams: LocalParams = LocalParams(remoteNodeId, DeterministicWallet.KeyPath(Seq(42L)), 1 sat, Long.MaxValue.msat, Some(500 sat), 1 msat, CltvExpiryDelta(144), 50, isInitiator = false, None, None, Features.empty)
+  val defaultParams: DefaultParams = DefaultParams(100 sat, 100000 msat, 100 msat, CltvExpiryDelta(288), 10)
+  val channels: Map[Peer.FinalChannelId, actor.ActorRef] = Map(Peer.FinalChannelId(randomBytes32()) -> system.classicSystem.deadLetters)
+  val connectedData: Peer.ConnectedData = Peer.ConnectedData(NodeAddress.fromParts("1.2.3.4", 42000).get, system.classicSystem.deadLetters, Init(TestConstants.Alice.nodeParams.features.initFeatures()), Init(TestConstants.Bob.nodeParams.features.initFeatures()), channels.map { case (k: ChannelId, v) => (k, v) })
+  def publicKey(fill: Byte): Crypto.PublicKey = PrivateKey(ByteVector.fill(32)(fill)).publicKey
```

Comment:
Most of this code is unused, I think you only need the following:

```suggestion
val remoteNodeId: Crypto.PublicKey = randomKey().publicKey
  val openChannel: OpenChannel = OpenChannel(ByteVector32.Zeroes, randomBytes32(), 250_000 sat, 0 msat, 1 sat, UInt64(1), 1 sat, 1 msat, FeeratePerKw(1 sat), CltvExpiryDelta(1), 1, randomKey().publicKey, randomKey().publicKey, randomKey().publicKey, randomKey().publicKey, randomKey().publicKey, randomKey().publicKey, ChannelFlags.Private)
  val defaultParams = DefaultParams(100 sat, 100_000 msat, 100 msat, CltvExpiryDelta(288), 10)
```

In particular, the `connectedData` was only used to access `localFeatures` and `remoteFeatures`. We should instead let each test specify what features are used: in most cases it will just be `Features.empty` for both as it has no impact on the test, except when you want to test that dual funding is used, where you'll explicitly add the dual funding feature in the corresponding tests.

Reply:
Good point! in fact, `openChannel` can be replaced by `createOpenChannelMessage`. Fixed in [522019b](https://github.com/ACINQ/eclair/pull/2552/commits/522019bb22f2ff980a173707db0ad7731583054b).